src/component/tooltip/TooltipHTMLContent.ts

```diff
@@ -350,7 +361,7 @@ class TooltipHTMLContent {
     update(tooltipModel: Model<TooltipOption>) {
         // FIXME
         // Move this logic to ec main?
-        const container = this._container;
+        const container = this._api.getDom();
```

Comment:
Can you explain why you changed the `_container` to `api.getDom()`? I think it should be unchanged.

Reply:
Sry. You are right.

Reply:
Agree with you. The tooltip probably can't position correctly if its container `position` isn't `relative`, but the user should be responsible for this risk when they decide to append the tooltip element to a customized container that is uncontrolled by the chart. 

So, maybe we can set the `position` only when the customized container is not specified?

``` ts
const container = this._container;
if (container === this._api.getDom()) {
  // ...
}
```